catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/MetacardVersionImpl.java

```diff
@@ -58,206 +57,83 @@
-
-    public static final String SKIP_VERSIONING = "skip-versioning";
-
-    public static final String HISTORY_METACARDS_PROPERTY = "history-metacards";
-
-    ////////////////////////////////////////////////////////////////////////////////////
-    // ATTRIBUTE VALUES
-    ////////////////////////////////////////////////////////////////////////////////////
-
-    /**
-     * {@link ddf.catalog.data.Attribute} value for {@link ddf.catalog.data.Metacard#TAGS} when
-     * a metacard is a History Metacard.
-     */
-    public static final String VERSION_TAG = "revision";
-
-    ////////////////////////////////////////////////////////////////////////////////////
-    // ATTRIBUTE NAMES
-    ////////////////////////////////////////////////////////////////////////////////////
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for action of the current {@link MetacardVersion}.
-     * Can be one of <code>Created</code>, <code>Updated</code>, or <code>Deleted</code>.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String ACTION = PREFIX.apply("action");
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for the editor of this {@link Metacard} revision.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String EDITED_BY = PREFIX.apply("edited-by");
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for version date of this {@link Metacard} revision.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String VERSIONED_ON = PREFIX.apply("versioned-on");
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for metacard ID on a history item of this {@link Metacard} revision.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String VERSION_OF_ID = PREFIX.apply("id");
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for original tags of this {@link Metacard} revision.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String VERSION_TAGS = PREFIX.apply("tags");
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for original metacard type of this {@link Metacard} revision.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String VERSION_TYPE = PREFIX.apply("type");
-
-    /**
-     * {@link ddf.catalog.data.Attribute} name for original serialized metacard type  of this {@link Metacard} revision.
-     *
-     * @since DDF-2.9.0
-     */
-    public static final String VERSION_TYPE_BINARY = PREFIX.apply("type-binary");
+    private static final Set<AttributeDescriptor> VERSION_DESCRIPTORS =
+            new HashSet<>(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
 
     static {
         VERSION_DESCRIPTORS.add(new AttributeDescriptorImpl(ACTION,
-                true /* indexed */,
-                true /* stored */,
-                false /* tokenized */,
-                false /* multivalued */,
+                true,
```

Comment:
Once  you update to intelliJ 2016.3 you will see. :D (spoiler alert: they put the names of ambiguous variables next to them inside of a method call)